### PR TITLE
update v1 -> v3 api

### PR DESCRIPTION
**Pull request by ddiaz-duplo**
## Change description

1. Update to v3 api which returns a more accurate `status_code` and `message`
2. Remove redundant pointers to slices as they are already passed by reference. This improves readability.
3. Replace empty slice declarations with nil slice declarations.

tested on test19.duplocloud.net with the following resource:

```
resource "duplocloud_tenant" "myapp" {
  account_name = "newtenant"
  plan_id      = "default"
}

resource "duplocloud_tenant_config" "myapp" {
  tenant_id = duplocloud_tenant.myapp.tenant_id

  // This turns on a default public access block for S3 buckets.
  setting {
    key   = "block_public_access_to_s3"
    value = "false"
  }

  // This turns on a default SSL enforcement S3 bucket policy.
  setting {
    key   = "enforce_ssl_for_s3"
    value = "false"
  }
}
```

image

## Type of change
- [ ] Bug fix (fixes an issue)
- [x] New feature (adds functionality)

### Development

- [x] Pull requests may not be submitted to the *master* branch (use *develop* instead) - or they will be closed.
- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [x] Pull request linked to task tracker where applicable

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-11479 TF: Replace v1 adminproxy/GetTenant with v3/admin/tenant/{t_id}/metadata](https://app.clickup.com/t/8655600/DUPLO-11479)

File: duplosdk/tenant.go [hunk 1]
```diff
@@ -142,18 +143,19 @@ func (c *Client) TenantDelete(tenantID string) ClientError {
 }
 
 // ListTenantsForUser retrieves a list of tenants for the current user via the Duplo API.
-func (c *Client) ListTenantsForUser() (*[]DuploTenant, ClientError) {
-	list := []DuploTenant{}
+func (c *Client) ListTenantsForUser() ([]DuploTenant, ClientError) {
+	var list []DuploTenant
```
Comment (joek-duplo) [inline duplosdk/tenant.go:147]: revert this, since it makes the `list` initially `nil`

File: duplosdk/tenant.go [hunk 2]
```diff
@@ -203,20 +205,20 @@ func (c *Client) GetTenantForUser(tenantID string) (*DuploTenant, ClientError) {
...
 	// No tenant was found.
-	return nil, nil
+	return nil, clientError{status: http.StatusNotFound, message: fmt.Sprintf("Tenant %s was not found", tenantID)}
 }
 
 // TenantGetConfig retrieves tenant configuration metadata via the Duplo API.
 func (c *Client) TenantGetConfig(tenantID string) (*DuploTenantConfig, ClientError) {
-	list := []DuploKeyStringValue{}
-	err := c.getAPI(fmt.Sprintf("TenantGetConfig(%s)", tenantID), fmt.Sprintf("adminproxy/GetTenantMetadata/%s", tenantID), &list)
+	var list []DuploKeyStringValue
```
Comment (joek-duplo) [inline duplosdk/tenant.go:220]: revert this, since it makes the `list` initially `nil`

File: duplosdk/tenant.go [hunk 3]
```diff
@@ -293,14 +295,14 @@ func (c *Client) TenantGetAwsCredentials(tenantID string) (*DuploTenantAwsCreden
 
 // TenantGetInternalSubnets retrieves a list of the internal subnets for a tenant via the Duplo API.
 func (c *Client) TenantGetInternalSubnets(tenantID string) ([]string, ClientError) {
-	list := []string{}
+	var list []string
```
Comment (joek-duplo) [inline duplosdk/tenant.go:298]: revert this, since it makes the `list` initially `nil`

File: duplosdk/tenant.go [hunk 4]
```diff
@@ -293,14 +295,14 @@ func (c *Client) TenantGetAwsCredentials(tenantID string) (*DuploTenantAwsCreden
...
 	err := c.getAPI(fmt.Sprintf("TenantGetInternalSubnets(%s)", tenantID), fmt.Sprintf("subscriptions/%s/GetInternalSubnets", tenantID), &list)
 	return list, err
 }
 
 // TenantGetExternalSubnets retrieves a list of the internal subnets for a tenant via the Duplo API.
 func (c *Client) TenantGetExternalSubnets(tenantID string) ([]string, ClientError) {
-	list := []string{}
+	var list []string
```
Comment (joek-duplo) [inline duplosdk/tenant.go:305]: revert this, since it makes the `list` initially `nil`

File: duplosdk/tenant.go [hunk 5]
```diff
@@ -371,7 +373,7 @@ func (c *Client) GetTenantEksSecret(tenantID string) (*DuploTenantEksSecret, Cli
 
 // TenantGetExtConnSecurityGroupRules retrieves a list of the external connection security group rules for a Duplo tenant.
 func (c *Client) TenantGetExtConnSecurityGroupRules(tenantID string) (*[]DuploTenantExtConnSecurityGroupRule, ClientError) {
-	list := []DuploTenantExtConnSecurityGroupRule{}
+	var list []DuploTenantExtConnSecurityGroupRule
```
Comment (joek-duplo) [inline duplosdk/tenant.go:376]: revert this, since it makes the `list` initially `nil`

File: duplosdk/tenant.go [hunk 6]
```diff
@@ -468,7 +470,7 @@ func (c *Client) TenantFeaturesGet(tenantId string) (*DuploTenantFeatures, Clien
 }
 
 func (c *Client) TenantTagsGet(tenantId string) (*[]DuploKeyStringValue, ClientError) {
-	rp := []DuploKeyStringValue{}
+	var rp []DuploKeyStringValue
```
Comment (joek-duplo) [inline duplosdk/tenant.go:473]: revert this, since it makes the `rp` initially `nil`

File: duplosdk/tenant.go [hunk 7]
```diff
@@ -477,7 +479,7 @@ func (c *Client) TenantTagsGet(tenantId string) (*[]DuploKeyStringValue, ClientE
 }
 
 func (c *Client) TenantTagsGetByKey(tenantId, key string) (*DuploKeyStringValue, ClientError) {
-	rp := []DuploKeyStringValue{}
+	var rp []DuploKeyStringValue
```
Comment (joek-duplo) [inline duplosdk/tenant.go:482]: revert this, since it makes the `rp` initially `nil`

**Review by joek-duplo — CHANGES_REQUESTED** [review]

**Review by joekhoobyar — APPROVED** [review]

**Review by joek-duplo — APPROVED** [review]

**Pull request merged**